### PR TITLE
fix #7740 test(nimbus): Extend targeting integration tests

### LINKED ISSUE #7740: [Desktop] extend tartgeting tests to use additional audience fields
┆Issue is synchronized with this [Jira Task](https://mozilla-hub.atlassian.net/browse/EXP-2785)

### PR DESCRIPTION
**Pull request by jrbenny35**
Because
* We currently check each targeting expression validates in the browser, but it is a very basic expression

This Commit
* Adds an addition test with additional fiends such as channel, locale, region and versions.

### CODE REVIEW

File: app/tests/integration/nimbus/test_firefox_targeting.py [hunk 1]
```diff
@@ -74,7 +74,7 @@ def targeting_config_slug(request):
 
 
 @pytest.mark.run_targeting
```
Comment (yashika-khurana) [inline app/tests/integration/nimbus/test_firefox_targeting.py:76]: do you think we should rename this to `run_advance_targeting`? 🤔  just to signify it contains more than required fields

**Review by yashika-khurana — CHANGES_REQUESTED** [review]
Just a minor suggestion, well done @jrbenny35  👏

File: app/tests/integration/nimbus/test_firefox_targeting.py (hunk 1 above)
Comment (jrbenny35) [reply app/tests/integration/nimbus/test_firefox_targeting.py:76]: Oh this marker is just for selecting tests when pytest runs. These tests should run with the old targeting tests.

Comment (yashika-khurana) [reply app/tests/integration/nimbus/test_firefox_targeting.py:76]: yup absolutely, I mean to say we are changing the meaning of marker from `targeting` to `advance targeting`, so would recommend refactoring that

Comment (jaredlockhart) [reply app/tests/integration/nimbus/test_firefox_targeting.py:76]: @yashika-khurana Yes I agree, however I think we should do a larger refactor of all the integration tests to more clearly separate and identify different testing concerns now that we've expanded the scope so much, so I'll file a followup for that with ideas for a better layout.

File: app/tests/integration/nimbus/test_firefox_targeting.py [hunk 2]
```diff
@@ -133,3 +133,82 @@ def test_check_targeting(
...
+@pytest.mark.parametrize(
+    "audience_field",
+    [
+        {"channel": "NIGHTLY"},
+        {"firefoxMinVersion": "FIREFOX_100"},
+        {"firefoxMaxVersion": "FIREFOX_120"},
+        {"locales": [37]},
+        {"countries": [42]},
+        {"proposedEnrollment": "14"},
+        {"proposedDuration": "30"},
```
Comment (jaredlockhart) [inline app/tests/integration/nimbus/test_firefox_targeting.py:147]: These don't go into targeting, so we don't need them here.

File: app/tests/integration/nimbus/test_firefox_targeting.py [hunk 3]
```diff
@@ -133,3 +133,82 @@ def test_check_targeting(
...
+                "description": "treatment branch",
+                "name": "Branch 2",
+                "ratio": 50,
+                "featureEnabled": False,
+                "featureValue": "",
+            }
+        ],
+        "populationPercent": "100",
+        "totalEnrolledClients": 55,
+    }
```
Comment (jaredlockhart) [inline app/tests/integration/nimbus/test_firefox_targeting.py:192]: We don't have like a fixture for this in conftest somewhere?

File: app/tests/integration/nimbus/test_firefox_targeting.py [hunk 4]
```diff
@@ -133,3 +133,82 @@ def test_check_targeting(
...
+    data.update(audience_field)
```
Comment (jaredlockhart) [inline app/tests/integration/nimbus/test_firefox_targeting.py:193]: Perf 👍

File: app/tests/integration/nimbus/test_firefox_targeting.py [hunk 5]
```diff
@@ -133,3 +133,82 @@ def test_check_targeting(
...
+    selenium.get("about:blank")
+    with open("nimbus/utils/filter_expression.js") as js:
+        result = Browser.execute_script(
+            selenium,
+            targeting,
+            json.dumps({"experiment": recipe}),
+            script=js.read(),
+            context="chrome",
+        )
+    assert result is not None, "Invalid Targeting, or bad recipe"
```
Comment (jaredlockhart) [inline app/tests/integration/nimbus/test_firefox_targeting.py:214]: I love how succinct we've gotten these 🙏

**Review by jaredlockhart — CHANGES_REQUESTED** [review]
Awesome thank you @jrbenny35 🙏 Just take those proposed durations out, and maybe there's a way to move that json fixture out?

File: app/tests/integration/nimbus/test_firefox_targeting.py (hunk 3 above)
Comment (jrbenny35) [reply app/tests/integration/nimbus/test_firefox_targeting.py:192]: Good idea, we can add that

**Review by jaredlockhart — APPROVED** [review]
Fantastic thank you @jrbenny35 🎉 🎉 🎉 🎉

**Pull request merged**